src/realm/cluster.cpp

```diff
@@ -810,7 +810,7 @@ size_t Cluster::erase(ObjKey key, CascadeState& state)
                         dict.remove_backlinks(state);
                     }
                 }
-                else if (col_type == col_type_LinkList) {
+                else if (col_type == col_type_LinkList || col_type == col_type_Link) {
```

Comment:
Clearly this works, but what's the difference between `col_type_Link` and   `col_type_TypedLink`

Reply:
We don't actually support `col_type_TypedLink` yet, so that is dead code right now.  TypedLink is currently only used internally as part of Mixed. The idea is that a column of type `col_type_Link` or `col_type_LinkList` can only link to a single static destination type, while a `TypedLink` has the flexibility to point to any type dynamically because it stores the destination object key and the destination object type. However, the issue in this PR is that `LnkLst` has type `col_type_LinkList` while `LnkSet` has type `col_type_Link` with the set attribute.